src/checks.py

```diff
@@ -215,7 +216,12 @@ def check_no_binary_files(state: State) -> Optional[str]:
 
 
 def check_build_and_test(state: State) -> Optional[str]:
-    return _check_sh("./mvnw --quiet package", workdir=state.source_dir)
+    return _check_sh(
+        "mvn --quiet -N io.takari:maven:wrapper -Dmaven=3.6.0",
+        "./mvnw --quiet test",
+        "./mvnw --quiet package",
```

Comment:
AFAIK `package` includes running tests, so  `test`  is redundant here (we'll run tests twice). Then again, I've been wrong about Maven before. Repeatedly.

Reply:
Yup, I was running them separately to more distinctly know where the failure happens when I did the test locally, I guess it doesn't really matter though, the output does tell you

Reply:
Fair enough. I don't have a strong preference here, go forth and merge as you like!

Reply:
I actually, now looking at the lifecycle docs think we might want to run verify, but it looks like that fails locally. I'm not sure we want to fail a release for integration test failures, at least if maybe they are known failures

Reply:
Pers'nally I think we're fine not running integration tests on release verification. Let's trust CI to take care of that, and only do a final sanity check here. Integration tests tend to be finicky and need much environment setup, which I don't think we should try to (a) 100% automate (b) put in the critical path on ASF releases. (Specifically these integration tests are known to fail in some local envs, but work fine on Jenkins)

Summoning @michaelsembwever to double-check that logic.

Reply:
See also the conversation starting at https://mail-archives.apache.org/mod_mbox/zipkin-dev/201902.mbox/browser - methinks there was weak consensus that we're OK not running integration tests at release vote verification.